### DISCUSSION TITLE
Why strictly depend on spidermonkey?

### DISCUSSION THREAD

**Discussion by sinm** [apache/couchdb#3499] [Ideas] [answered]
It's clear that JS-based designs are classic. But e.g. i'm using ruby backend/view-servers since 1.x and see no strict JS requirements but test cases & inline (?) generated JS-functions for the replication db.

There are many complications tied to spidermonkey under various os/arch combinations. E.g. python used by build, strange runtime bugs etc.

Would it be beautiful having configure-time argument that disables spidermonkey dependency leaving it to experienced users to define their view-servers?

**Comment by florianb** [comment]
Hi @sinm,

i can remember having read or asking about the same topic some a year ago (or so). In fact it was before the big update bumping SpiderMonkey to a recent version.

As far as i remember there is no "strict" dependency. SpiderMonkey is just the currently implemented interpreter which is delivered with the standard distribution (which has just historical reasons).

You are able to run CouchDb with custom Query-Servers: https://docs.couchdb.org/en/stable/config/query-servers.html

I personally don't know of current and up to date query server implementations, but its not anything more than providing the Query Server Protocol: https://docs.couchdb.org/en/latest/query-server/protocol.html with an executable.

I guess just somebody has to create another implementation.

**Reply by sinm** [reply]
As I mentioned previously I'm actually used to ruby query servers. The problem is you cant build or replicate w/o Spidermonkey.

**Reply by florianb** [reply]
I'm sorry - i missed that.

**Comment by wohali** [comment, marked as answer]
@sinm This is a duplicate of issue #1513 . Pull requests are welcome!

**Reply by sinm** [reply]
Ok, reading some Erlang books now, I'll be back. :)
